### PR TITLE
Free and transitive higher group actions

### PR DESCRIPTION
**Pull request by EgbertRijke**
In this PR we prove some basic consequences of the Regensburg extension of the fundamental theorem of identity types.

Part of this formalization was written during the live tutorial this morning.

### CODE REVIEW

File: src/foundation/surjective-maps.lagda.md [hunk 1]
```diff
@@ -257,17 +257,22 @@ module _
...
   abstract
     is-surjective-htpy :
-      f ~ g → is-surjective g → is-surjective f
-    is-surjective-htpy H K b =
+      {f g : A → B} → f ~ g → is-surjective g → is-surjective f
+    is-surjective-htpy {f} {g} H K b =
       apply-universal-property-trunc-Prop
         ( K b)
         ( trunc-Prop (fiber f b))
         ( λ {(a , refl) → unit-trunc-Prop (a , H a)})
```
Comment (fredrik-bakke) [inline src/foundation/surjective-maps.lagda.md:270]:
```suggestion
        ( λ where (a , refl) → unit-trunc-Prop (a , H a))
```
:)

File: src/higher-group-theory/free-higher-group-actions.lagda.md [hunk 2]
```diff
@@ -38,9 +64,90 @@ module _
...
+
+### The predicate of being an abstractly free ∞-group action
+
+```agda
+module _
+  {l1 l2 : Level} (G : ∞-Group l1) (X : action-∞-Group l2 G)
+  where
 
+  is-abstractly-free-action-∞-Group-Prop : Prop (l1 ⊔ l2)
+  is-abstractly-free-action-∞-Group-Prop =
```
Comment (fredrik-bakke) [inline src/higher-group-theory/free-higher-group-actions.lagda.md:77]: Shouldn't this be
```suggestion
  is-abstractly-free-action-prop-∞-Group : Prop (l1 ⊔ l2)
  is-abstractly-free-action-prop-∞-Group =
```
or
```suggestion
  is-abstractly-free-prop-action-∞-Group : Prop (l1 ⊔ l2)
  is-abstractly-free-prop-action-∞-Group =
```

File: src/higher-group-theory/free-higher-group-actions.lagda.md [hunk 3]
```diff
@@ -38,9 +64,90 @@ module _
...
+    is-emb-tr-is-abstractly-free-action-∞-Group :
+      is-abstractly-free-action-∞-Group G X →
+      (u : classifying-type-∞-Group G)
+      (x : type-action-∞-Group G X) →
+      is-emb (λ (p : shape-∞-Group G ＝ u) → tr X p x)
+    is-emb-tr-is-abstractly-free-action-∞-Group H u x =
+      apply-universal-property-trunc-Prop
+        ( mere-eq-classifying-type-∞-Group G (shape-∞-Group G) u)
+        ( is-emb-Prop _)
+        ( λ { refl → H x})
```
Comment (fredrik-bakke) [inline src/higher-group-theory/free-higher-group-actions.lagda.md:115]:
```suggestion
        ( λ where refl → H x)
```

File: src/higher-group-theory/transitive-higher-group-actions.lagda.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+  is-transitive-action-∞-Group : UU (l1 ⊔ l2)
+  is-transitive-action-∞-Group = type-Prop is-transitive-prop-action-∞-Group
+
+  is-prop-is-transitive-action-∞-Group : is-prop is-transitive-action-∞-Group
+  is-prop-is-transitive-action-∞-Group =
+    is-prop-type-Prop is-transitive-prop-action-∞-Group
+```
+
+### The predicate of bein an abstractly transitive higher group action
```
Comment (fredrik-bakke) [inline src/higher-group-theory/transitive-higher-group-actions.lagda.md:77]:
```suggestion
### The predicate of being an abstractly transitive higher group action
```

File: src/higher-group-theory/transitive-higher-group-actions.lagda.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+    is-surjective-tr-is-abstractly-transitive-action-∞-Group :
+      is-abstractly-transitive-action-∞-Group G X →
+      (u : classifying-type-∞-Group G)
+      (x : X (shape-∞-Group G)) →
+      is-surjective (λ (p : shape-∞-Group G ＝ u) → tr X p x)
+    is-surjective-tr-is-abstractly-transitive-action-∞-Group H u x =
+      apply-universal-property-trunc-Prop
+        ( mere-eq-classifying-type-∞-Group G (shape-∞-Group G) u)
+        ( is-surjective-Prop _)
+        ( λ { refl → pr2 H x})
```
Comment (fredrik-bakke) [inline src/higher-group-theory/transitive-higher-group-actions.lagda.md:130]:
```suggestion
        ( λ where refl → pr2 H x)
```

**Review by fredrik-bakke — APPROVED** [review]

File: src/foundation/surjective-maps.lagda.md (hunk 1 above)
Comment (EgbertRijke) [reply src/foundation/surjective-maps.lagda.md:270]: Your script is finding them already!:)

Comment (fredrik-bakke) [reply src/foundation/surjective-maps.lagda.md]: Hahaha, and I haven't even written it yet!

**Pull request merged**